src/wallet/wallet.h

```diff
@@ -145,6 +149,19 @@ class CKeyPool
             READWRITE(nVersion);
         READWRITE(nTime);
         READWRITE(vchPubKey);
+        if (ser_action.ForRead()) {
+            try {
+                READWRITE(fInternal);
+            }
+            catch (std::ios_base::failure&) {
+                /* flag as external address if we can't read the internal boolean
+                   (this will be the case for any wallet before the HD chain split version) */
+                fInternal = false;
```

Comment:
Instead of doing this nasty nasty thing, bump the version and add an if here.

Reply:
bump what version exactly?